### DISCUSSION TITLE
Get code warnings with fastlane

### DISCUSSION THREAD

**Discussion by revolter** [fastlane/fastlane#17460] [Q&A]
Is there a way to retrieve the code warnings using fastlane? Maybe with `gym`. And then feed them somehow to a CI running from a PR.

**Comment by rogerluan** [comment]
I don't recall any way to do this using just _fastlane_ to be honest. But you could inspect how SwiftInfo does it (or use it): https://github.com/rockbruno/SwiftInfo - spoiler: it parses `xcodebuild` build log file

Hope this helps!

**Reply by revolter** [reply]
I found out [danger-xcodebuild](https://github.com/valeriomazzeo/danger-xcodebuild) and [danger-xcode_summary](https://github.com/diogot/danger-xcode_summary) but didn't have time to test them yet.

**Reply by rogerluan** [reply]
Nice. I also noticed that the way that SwiftInfo does (which I haven't look into how it does it) is not perfect. It gets some of the warnings, but definitely not custom ones (e.g. build warnings generated by SwiftLint).

I might try those 2 options out to see if they list them correctly
